pkg/xds/generator/probe_generator.go

```diff
@@ -0,0 +1,52 @@
+package generator
+
+import (
+	"github.com/pkg/errors"
+
+	model "github.com/kumahq/kuma/pkg/core/xds"
+	xds_context "github.com/kumahq/kuma/pkg/xds/context"
+	envoy_listeners "github.com/kumahq/kuma/pkg/xds/envoy/listeners"
+	"github.com/kumahq/kuma/pkg/xds/envoy/names"
+	envoy_routes "github.com/kumahq/kuma/pkg/xds/envoy/routes"
+)
+
+const (
+	// OriginProbes is a marker to indicate by which ProxyGenerator resources were generated.
+	OriginProbe  = "probe"
+	listenerName = "probe:listener"
+)
+
+type ProbeProxyGenerator struct {
+}
+
+func (g ProbeProxyGenerator) Generate(ctx xds_context.Context, proxy *model.Proxy) (*model.ResourceSet, error) {
+	probes := proxy.Dataplane.Spec.Probes
```

Comment:
please place an if when Probes are nil to return nil immediately instead of generating listener that we don't user